### PR TITLE
feat: Allow additional login params, Introduce `LogInValidator`

### PR DESCRIPTION
**Pull request by imorland**
Precursor to https://discuss.flarum.org/d/31904-enabling-cloudflare-turnstile-on-login

**Changes proposed in this pull request:**
- Extract `params` into an extendable function so that additional params can be added in `LogInModal`
- `LogInController` runs validation of the supplied `params` against a new `LogInValidator` - empty ruleset by default.

**Reviewers should focus on:**

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

File: framework/core/js/src/forum/components/LogInModal.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import ItemList from '../../common/utils/ItemList';
 import Stream from '../../common/utils/Stream';
 import type Mithril from 'mithril';
 import RequestError from '../../common/utils/RequestError';
+import { LoginParams } from '../../common/Session';
```
Comment (davwheat) [inline framework/core/js/src/forum/components/LogInModal.tsx]:
```suggestion
import type { LoginParams } from '../../common/Session';
```

**Review by davwheat — APPROVED** [review]

**Review by SychO9 — COMMENTED** [review]
Hmm, an event doesn't really seem appropriate to me. I'd rather we create a login validator, empty by default, and extensions can extend the validator instead.

**Comment by imorland** [comment]
🤔 yeah... That would make more sense, actually. On it...

**Comment by davwheat** [comment]
Why not both?

That would allow for processing of data from a user logging in, and the more traditional approach to validation.

**Comment by SychO9** [comment]
I don't know, it's not apparent to me what the processing of data could be, and if whatever it is it couldn't be handled using the exsiting extension API (like we did here with the validator), so I prefer dealing with one use case at a time to avoid over-engineering it.

**Review by SychO9 — APPROVED** [review]

**Pull request merged**